### PR TITLE
aarch64 executable tests

### PR DESCRIPTION
**Pull request by pfilipko1**
## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have updated the relevant documentation.
- [x] I have added tests for new logic.

### CODE REVIEW

**Comment by pfilipko1** [comment]
I had to install staticx manually as root before, but I think it is a matter of my testing environment, so I decided not to add this change to this PR. If the problem turns out to be aarch64 specific, I'll add this installation to the tests/test.sh file.

**Comment by Jongy** [comment]
It doesn't make sense to me that specifically the "not_privileged" tests will fail on Aarch64, this is most likely something with the environment

File: tests/conftest.py [hunk 1]
```diff
@@ -186,18 +186,8 @@ def application_executable(runtime: str) -> str:
 @fixture
 def gprofiler_exe(request: FixtureRequest, tmp_path: Path) -> Path:
     precompiled = request.config.getoption("--executable")
-    if precompiled is not None:
-        return Path(precompiled)
-
-    with chdir(PARENT):
```
Comment (Jongy) [inline tests/conftest.py:192]: For the record, we are removing it here bcause this diverges from the pyi.Dockerfile. We follow the dockerfile build, not the logic here.

**Comment by Jongy** [comment]
I ran this on my Aarch64 server:
```
sudo ./tests/test.sh --executable build/aarch64/gprofiler
```
and got:
```
=========================================================== 184 passed, 26 skipped, 27 xfailed, 676 warnings in 2397.70s (0:39:57) ===========================================================
```

Will merge this soon

**Comment by Jongy** [comment]
Well, the Aarch64 executable pyperf test is marked as passing
```
tests/test_executable.py::test_executable[False-python-pyperf] PASSED                                                                                                                  [ 18%]
```

because `PythonProfiler` implicitly falls back to py-spy despite pyperf being requested, this needs to be fixed + the test should check for kernel + native stacks like the container test. I'll fix in a separate PR.

**Pull request merged**